spec/factories/events.rb

```diff
@@ -1,8 +1,9 @@
 FactoryBot.define do
   factory :event do
-    association(:eventable)
+    eventable { association(:move) }
     event_name { 'create' }
     client_timestamp { Time.now.utc + rand(-60..60).seconds } # NB: the client_timestamp will never be perfectly in sync with system clock
+    details { { supplier_id: '1234', event_params: { attributes: { notes: 'foo' } }, data_params: { attributes: { notes: 'bar' } } } }
```

Comment:
This formatting is pretty hard to read - perhaps break over multiple lines inside a `do .. end` block. Same for line 42.